test/e2e/predictor/test_torchserve.py

```diff
@@ -37,24 +39,108 @@ def test_torchserve_kserve():
             storage_uri="gs://kfserving-examples/models/torchserve/image_classifier",
             protocol_version="v1",
             resources=V1ResourceRequirements(
-                requests={"cpu": "1", "memory": "4Gi"},
-                limits={"cpu": "1", "memory": "4Gi"},
+                requests={
+                    "cpu": "1",
+                    "memory": "4Gi"
+                },
+                limits={
+                    "cpu": "1",
+                    "memory": "4Gi"
+                },
             ),
         ),
     )
 
     isvc = V1beta1InferenceService(
         api_version=constants.KSERVE_V1BETA1,
         kind=constants.KSERVE_KIND,
-        metadata=client.V1ObjectMeta(
-            name=service_name, namespace=KSERVE_TEST_NAMESPACE
-        ),
+        metadata=client.V1ObjectMeta(name=service_name, namespace=KSERVE_TEST_NAMESPACE),
         spec=V1beta1InferenceServiceSpec(predictor=predictor),
     )
 
     kserve_client.create(isvc)
     kserve_client.wait_isvc_ready(service_name, namespace=KSERVE_TEST_NAMESPACE)
 
     res = predict(service_name, "./data/torchserve_input.json")
-    assert(res.get("predictions")[0] == 2)
+    assert (res.get("predictions")[0] == 2)
+    kserve_client.delete(service_name, KSERVE_TEST_NAMESPACE)
+
+
+def test_torchserve_v2_kserve():
+    service_name = "mnist-v2"
+    predictor = V1beta1PredictorSpec(
+        min_replicas=1,
+        pytorch=V1beta1TorchServeSpec(
+            storage_uri="gs://kfserving-examples/models/torchserve/image_classifier/v2",
+            protocol_version="v2",
+            resources=V1ResourceRequirements(
+                requests={
+                    "cpu": "1",
+                    "memory": "4Gi"
+                },
+                limits={
+                    "cpu": "1",
+                    "memory": "4Gi"
+                },
+            ),
+        ),
+    )
+
+    isvc = V1beta1InferenceService(
+        api_version=constants.KSERVE_V1BETA1,
+        kind=constants.KSERVE_KIND,
+        metadata=client.V1ObjectMeta(name=service_name, namespace=KSERVE_TEST_NAMESPACE),
+        spec=V1beta1InferenceServiceSpec(predictor=predictor),
+    )
+
+    kserve_client.create(isvc)
+    kserve_client.wait_isvc_ready(service_name, namespace=KSERVE_TEST_NAMESPACE)
+
+    res = predict(service_name, "./data/torchserve_input_v2.json", model_name="mnist")
+    assert (res.get("outputs")[0]["data"] == [1])
+    kserve_client.delete(service_name, KSERVE_TEST_NAMESPACE)
+
+
+def test_torchserve_grpc():
+    service_name = "mnist"
+    predictor = V1beta1PredictorSpec(
+        min_replicas=1,
+        pytorch=V1beta1TorchServeSpec(
+            storage_uri="gs://kfserving-examples/models/torchserve/image_classifier/",
+            ports=[V1ContainerPort(container_port=7070, name="h2c", protocol="TCP")],
```

Comment:
where is 7070 configured ? I do not find it anywhere in the configuration file

Reply:
It is the default port for Torchserve gRPC